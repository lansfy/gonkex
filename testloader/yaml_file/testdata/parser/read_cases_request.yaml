- name: loader MUST substitute requestArgs to requests components
  method: POST
  path: /some/path/{{ .foo }}
  request: "{{ .foo }} - {{ .bar }} {{ $var }}"
  headers:
    X-Header: "{{ .foo }}"
  cookies:
    "Cookie": "{{ .bar }}"
  query: ?{{ .bar }}={{ .foo }}
  response:
    200: "{{ .foo }} - not affected by requestArgs"
  cases:
    - requestArgs:
        foo: 'value 1'
        bar: other1
    - requestArgs:
        foo: 'value 2'
        bar: other2
