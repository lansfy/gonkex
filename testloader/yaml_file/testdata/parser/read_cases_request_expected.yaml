- GetName: "loader MUST substitute requestArgs to requests components #1"
  GetMethod: POST
  Headers:
    X-Header: value 1
  Cookies:
    Cookie: other1
  GetRequest: "value 1 - other1 {{ $var }}"
  Path: /some/path/value 1
  ToQuery: ?other1=value 1
  GetResponses:
    200: "{{ .foo }} - not affected by requestArgs"
  GetFileName: testdata/parser/read_cases_request.yaml
  OneOfCase: true
  FirstTestInFile: true

- GetName: "loader MUST substitute requestArgs to requests components #2"
  GetMethod: POST
  Headers:
    X-Header: value 2
  Cookies:
    Cookie: other2
  Path: /some/path/value 2
  ToQuery: ?other2=value 2
  GetRequest: "value 2 - other2 {{ $var }}"
  GetResponses:
    200: "{{ .foo }} - not affected by requestArgs"
  GetFileName: testdata/parser/read_cases_request.yaml
  OneOfCase: true
  LastTestInFile: true
